src/service.ts

```diff
@@ -189,19 +204,13 @@ export class Service {
         ]
       };
       const input = encodeURIComponent(JSON.stringify(project)).replace("%20", "+");
-      return this.sendRequest("input=" + input + "&action=build");
+      return this.sendRequest("input=" + input + "&action=build", "service");
     } else if (from === Language.Wasm && to === Language.x86) {
       const input = encodeURIComponent(base64js.fromByteArray(src as ArrayBuffer));
-      return this.sendRequest("input=" + input + "&action=wasm2assembly&options=" + encodeURIComponent(options));
+      return this.sendRequest("input=" + input + "&action=wasm2assembly&options=" + encodeURIComponent(options), "service");
     } else if (from === Language.Rust && to === Language.Wasm) {
       // TODO: Temporary until we integrate rustc into the service.
-      return fetch("https://rust-heroku.herokuapp.com/rustc", {
-        method: "POST",
-        body: JSON.stringify({code: src}),
-        headers: new Headers({
-          "Content-Type": "application/json"
-        })
-      }).then(res => res.json());
+      return this.sendRequest(JSON.stringify({code: src}), "rustc");
```

Comment:
It's hard to follow what `sendRequest` returns, in this case an object expected, however it was changed to return string.

Can you:
 - remove inaccurate refactoring and keep it as it was before
 - or, introduce sendRequestJSON to handle a new way of communicating with service?

Reply:
`sendRequest` return `return JSON.parse(await response.text());` which will be an object